static/js/components/forms/CouponForm.js

```diff
@@ -0,0 +1,298 @@
+                </label>
+                <ErrorMessage name="max_redemptions" component="div" />
+              </div>
+            </React.Fragment>
+          ) : (
+            <React.Fragment>
+              <label htmlFor="num_coupon_codes">
+                Number of coupon codes*
+                <Field name="num_coupon_codes" />
+              </label>
+              <ErrorMessage name="num_coupon_codes" component="div" />
+            </React.Fragment>
+          )}
+        </div>
+
+        <div>
+          <div>
+            <Field type="checkbox" name="automatic" />
+            Automatically apply coupon to eligible products in basket
+          </div>
+          <div className="block">
+            <label htmlFor="amount">
+              Percentage Discount (1 to 100)*
+              <Field name="amount" />
+            </label>
+            <ErrorMessage name="amount" component="div" />
+          </div>
+          <div className="block">
+            <label htmlFor="name">
+              Unique Coupon Name (letters, numbers, underscore only)*
+              <Field name="name" />
+            </label>
+            <ErrorMessage name="name" component="div" />
+          </div>
+          <div className="block">
+            <label htmlFor="tag">
+              Tag (optional)
+              <Field name="tag" />
+            </label>
+          </div>
+        </div>
+        <div className="flex">
+          <div className="block">
+            <label htmlFor="activation_date">
+              Valid from*
+              <DayPickerInput
+                name="activation_date"
+                placeholder="MM/DD/YYYY"
+                format="MM/DD/YYYY"
+                onDayChange={value => {
+                  setFieldValue("activation_date", value)
+                }}
+                onDayPickerHide={() => setFieldTouched("activation_date")}
+                error={errors.activation_date}
+                touched={touched.activation_date}
+              />
+            </label>
+            <ErrorMessage name="activation_date" component="div" />
+          </div>
+          <div className="block">
+            <label htmlFor="expiration_date">
+              Valid until*
+              <DayPickerInput
+                name="expiration_date"
+                placeholder="DD/MM/YYYY"
+                format="DD/MM/YYYY"
+                onDayChange={value => {
+                  setFieldValue("expiration_date", value)
+                }}
+                onDayPickerHide={() => setFieldTouched("expiration_date")}
+                error={errors.expiration_date}
+                touched={touched.expiration_date}
+              />
+            </label>
+            <ErrorMessage name="expiration_date" component="div" />
+          </div>
+        </div>
+        <div className="flex">
+          <Field
+            type="radio"
+            name="product_type"
+            value="program"
+            onClick={toggleProduct}
```

Comment:
Same comment as above, I don't think you need to or should manage this state outside the form.